src/components/Api.js

```diff
@@ -0,0 +1,91 @@
+export default class Api {
+  constructor(options) {
+    this.server = options.baseUrl;
+    this.headers = options.headers;
+  }
+
+  _checkResponse(res) {
+    if (res.ok) {
+      return res.json();
+    }
+    return Promise.reject(`Error ${res.status}`);
+  }
+
+  getInitialCards() {
+    return fetch(`${this.server}/cards`, {
+      headers: this.headers,
+    })
+      .then(this._checkResponse)
+      .then((result) => {
```

Comment:
Needs correction: there is no need to have this `then` block as it does nothing with the response (please fix that everywhere)